src/main/resources/assets/react/documentTypes/NewOrEditDocumentType.jsx

```diff
@@ -134,121 +172,127 @@ export function NewOrEditDocumentType({
-								addFields,
-								properties*/
-							} = json.data.createDocumentType;
-							/*setInitialValues({ // So reset button doesn't empty all inputs
-								_name,
-								addFields,
-								properties
-							});*/
-							setId(_id);
-							setInitialValues(false); // Should unmount the EnonicForm, trigger getDocumentType, and remount Enonicform?
-						});
-					}
+				const variables = {
+					_name,
+					addFields,
+					properties
+				};
+				if (_id) {
+					variables._id = _id;
+					variables._versionKey = initialValues._versionKey;
 				}
-			});
-		}}
-		schema={SCHEMA}
-	>
-		<Modal.Content>
-			<Tab
-				defaultActiveIndex={0}
-				panes={(() => {
-					const panes = [];
-					if (_id) {
+				//console.debug('submit variables', variables);
+
+				fetch(`${servicesBaseUrl}/graphQL`, {
+					method: 'POST',
+					headers: {
+						'Content-Type':	'application/json'
+					},
+					body: JSON.stringify({
+						query: _id ? GQL_MUTATION_DOCUMENT_TYPE_UPDATE : GQL_MUTATION_DOCUMENT_TYPE_CREATE,
+						variables
+					})
+				}).then(response => {
+					//console.debug('response', response);
+					if (response.status === 200) {
+						if (_id) {
+							doClose();
+						} else {
+							//console.debug('response.json()', response.json()); // Promise
+							response.json().then(json => {
+								//console.debug('json', json);
+								const {
+									_id/*,
+									_name,
+									addFields,
+									properties*/
+								} = json.data.createDocumentType;
+								/*setInitialValues({ // So reset button doesn't empty all inputs
+									_name,
+									addFields,
+									properties
+								});*/
+								setId(_id);
+								setInitialValues(false); // Should unmount the EnonicForm, trigger getDocumentType, and remount Enonicform?
+							});
+						}
+					}
+				});
+			}}
+			schema={SCHEMA}
+		>
+			<Modal.Content>
+				<Tab
+					defaultActiveIndex={0}
+					panes={(() => {
+						const panes = [];
+						if (_id) {
+							panes.push({
+								menuItem: {
+									content: 'Fields',
+									icon: 'list',
+									key: 'fields'
+								},
+								render: () => <Tab.Pane>
+									<FieldsList
+										collectionsArr={collectionsArr}
+										globalFields={globalFields}
+										interfacesArr={interfacesArr}
+										servicesBaseUrl={servicesBaseUrl}
+									/>
+								</Tab.Pane>
+							});
+						} // if _id
 						panes.push({
 							menuItem: {
-								content: 'Fields',
-								icon: 'list',
-								key: 'fields'
+								content: 'Settings',
+								icon: 'setting',
+								key: 'settings'
 							},
 							render: () => <Tab.Pane>
-								<FieldsList
-									collectionsArr={collectionsArr}
-									globalFields={globalFields}
-									interfacesArr={interfacesArr}
-									servicesBaseUrl={servicesBaseUrl}
-								/>
+								<Form as='div'>
+									<Form.Field>
+										<Input
+											value={nameInput}
+											onInput={e => {setNameInput(e.target.value);}}
```

Comment:
So, I'm not too happy with this. It's normal to use onChange not onInput. But the thing is the `<Input>` you are changing is not a normal `<input>` nor a normal `<SemanticUiReactInput>`, but it's acutally a `<SemanticUiReactFormInput>` which has it's own onChange, because it's an input to be used inside a `<SemanticUiReactForm>`. It has it's own validation, nested state, error and touched state and more.

If you need/want to roll you'r own onChange, it's better just to use a normal `<SemanticUiReactInput>` and then remember to use it's state value when submitting the form, rather that the nested state from semantic-ui-react-form...